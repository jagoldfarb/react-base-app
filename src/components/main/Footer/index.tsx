import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';

import { SOCIAL_MEDIAS } from 'helpers/enums/socialMedias';

import Logo from 'components/commons/Logo';

import {
  FooterContainer,
  FooterContent,
  LogoContainer,
  Copyright,
  InfoContainer,
  LinksContainer,
  SocialMediasContainer,
  Facebook,
  Instagram,
  LinkedIn,
} from './styles';

const Footer: FunctionComponent = () => (
  <FooterContainer data-testid="footer">
    <FooterContent>
      <LogoContainer>
        <Logo isDarkModeOn />
        <Copyright>&copy; Copyright {new Date().getFullYear()}</Copyright>
      </LogoContainer>
      <InfoContainer>
        <LinksContainer data-testid="footer-links-container">
          <li>
            <Link to="/link-1">Link 1</Link>
          </li>
          <li>
            <Link to="/link-2">Link 2</Link>
          </li>
          <li>
            <Link to="/link-3">Link 3</Link>
          </li>
        </LinksContainer>
        <SocialMediasContainer data-testid="footer-social-medias-container">
          <a target="_blank" rel="noopener noreferrer" href={SOCIAL_MEDIAS.linkedIn.link}>
            <LinkedIn />
          </a>
          <a target="_blank" rel="noopener noreferrer" href={SOCIAL_MEDIAS.instagram.link}>
            <Instagram />
          </a>
          <a target="_blank" rel="noopener noreferrer" href={SOCIAL_MEDIAS.facebook.link}>
            <Facebook />
          </a>
        </SocialMediasContainer>
      </InfoContainer>
    </FooterContent>
  </FooterContainer>
);

export default Footer;
