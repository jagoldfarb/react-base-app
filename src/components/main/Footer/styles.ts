import styled, { css } from 'styled-components';

import { FOOTER_HEIGHT } from 'helpers/dimensions';
import { COLORS } from 'helpers/enums/colors';
import { SIZES } from 'helpers/enums/sizes';

import { ReactComponent as FacebookIcon } from './assets/FacebookIcon.svg';
import { ReactComponent as LinkedInIcon } from './assets/LinkedInIcon.svg';
import { ReactComponent as InstagramIcon } from './assets/InstagramIcon.svg';

export const FooterContainer = styled.footer`
  background-color: ${COLORS.black};
  width: 100%;
  height: ${FOOTER_HEIGHT}px;

  @media (max-width: ${SIZES.mediumPhone}px) {
    height: auto;
  }
`;

export const FooterContent = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  max-width: ${SIZES.smallDesktop}px;
  margin: 0 auto;
  padding-top: 30px;
  color: ${COLORS.white};

  @media (max-width: ${SIZES.smallDesktop}px) {
    padding: 0 25px;
    padding-top: 30px;
  }

  @media (max-width: ${SIZES.mediumPhone}px) {
    flex-direction: column;
  }
`;

export const LogoContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Copyright = styled.p`
  margin-top: 17px;
  font-size: 14px;
`;

export const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const LinksContainer = styled.ul`
  display: flex;
  flex-direction: row;
  list-style-type: none;
  margin: 0;
  padding: 0;

  li {
    padding-right: 50px;
  }

  a {
    color: inherit;
  }

  li:last-child {
    padding-right: 0;
  }

  @media (max-width: ${SIZES.mediumPhone}px) {
    flex-direction: column;
    margin-top: 20px;

    li {
      padding-right: 0;
      padding-bottom: 30px;
    }
  }
`;

export const SocialMediasContainer = styled.div`
  display: flex;
  flex-direction: row;

  @media (max-width: ${SIZES.mediumPhone}px) {
    padding-bottom: 20px;
  }
`;

export const SocialMediaIcon = css`
  height: 26px;
  width: 26px;
  fill: ${COLORS.white};
  margin-right: 10px;
`;

export const Facebook = styled(FacebookIcon)`
  ${SocialMediaIcon}
`;

export const Instagram = styled(InstagramIcon)`
  ${SocialMediaIcon}
`;

export const LinkedIn = styled(LinkedInIcon)`
  ${SocialMediaIcon}
`;
