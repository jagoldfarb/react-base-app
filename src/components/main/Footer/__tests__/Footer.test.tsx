import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Footer from '..';

describe('Footer Component', () => {
  it('Should render the footer', () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>,
    );
    expect(getByTestId('footer')).toBeDefined();
    expect(getByTestId('footer-links-container')).toBeDefined();
    expect(getByTestId('footer-social-medias-container')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
