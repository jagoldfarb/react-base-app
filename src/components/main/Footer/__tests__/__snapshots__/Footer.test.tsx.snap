// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should render the footer 1`] = `
<DocumentFragment>
  <footer
    class="sc-gsTCUz fkwGEN"
    data-testid="footer"
  >
    <div
      class="sc-dlfnbm bcMIIj"
    >
      <div
        class="sc-hKgILt jsOJFj"
      >
        <div
          class="sc-bdfBwQ kCPbJH"
          data-testid="logo"
        >
          <a
            href="/home"
          >
            Logo
          </a>
        </div>
        <p
          class="sc-eCssSg JMFhk"
        >
          Â© Copyright 2021
        </p>
      </div>
      <div
        class="sc-jSgupP dqEKVT"
      >
        <ul
          class="sc-gKsewC cGHNYq"
          data-testid="footer-links-container"
        >
          <li>
            <a
              href="/link-1"
            >
              Link 1
            </a>
          </li>
          <li>
            <a
              href="/link-2"
            >
              Link 2
            </a>
          </li>
          <li>
            <a
              href="/link-3"
            >
              Link 3
            </a>
          </li>
        </ul>
        <div
          class="sc-iBPRYJ dAtxtZ"
          data-testid="footer-social-medias-container"
        >
          <a
            href="https://www.linkedin.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="sc-jrAGrp htObCd"
            >
              LinkedInIcon.svg
            </svg>
          </a>
          <a
            href="https://www.instagram.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="sc-pFZIQ jSCRPh"
            >
              InstagramIcon.svg
            </svg>
          </a>
          <a
            href="https://www.facebook.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="sc-fubCfw bnodce"
            >
              FacebookIcon.svg
            </svg>
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
